aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3427,8 +3429,30 @@ static void linalg_lstsq_out_info(
   auto input_working_copy = copyBatchedColumnMajor(input);
 
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if (driver == "gelss" && input.device() != at::kCPU) {
+    if (input.numel() == 0) {
+        auto output_shape = input.sizes().vec();
+        output_shape.back() = other.size(-1);
+        solution.zero_();
+    } else {
+        auto [U, S, Vh] = at::_linalg_svd(input, false, true, "gesvd");
+        rank = at::zeros({1}, at::kLong);
+
+        auto S_pinv = S.reciprocal();
+        auto s1 = at::narrow(S, /*dim=*/-1, /*start=*/0, /*length=*/1);  // singular values are sorted in descending order
+        S_pinv.masked_fill_(S < rcond * s1, 0);
+        rank[0] = (S != 0).sum();
```

Comment:
This is not correct. Please see the definition of what is considered a zero singular value in the docs of this function.

Reply:
I meant to use the pseudo inverse here, not `S`, that is my bad.